src/aind_metadata_mapper/open_ephys/utils/behavior_utils.py

```diff
@@ -0,0 +1,1479 @@
+    stimulus_table = stimulus_table[sorted(stimulus_table.columns)]
+    return stimulus_table
+
+
+def get_images_dict(pkl_dict) -> Dict:
+    """
+    Gets the dictionary of images that were presented during an experiment
+    along with image set metadata and the image specific metadata. This
+    function uses the path to the image pkl file to read the images and their
+    metadata from the pkl file and return this dictionary.
+    Parameters
+    ----------
+    pkl_dict: The pkl file containing the data for the stimuli presented during
+         experiment
+
+    Returns
+    -------
+    Dict:
+        A dictionary containing keys images, metadata, and image_attributes.
+        These correspond to paths to image arrays presented, metadata
+        on the whole set of images, and metadata on specific images,
+        respectively.
+
+    """
+    # Sometimes the source is a zipped pickle:
+    pkl_stimuli = pkl_dict["items"]["behavior"]["stimuli"]
+    metadata = {"image_set": pkl_stimuli["images"]["image_path"]}
+
+    # Get image file name;
+    # These are encoded case-insensitive in the pickle file :/
+    filename = stim.convert_filepath_caseinsensitive(metadata["image_set"])
+
+    image_set = pkl.load_img_pkl(open(filename, "rb"))
+    images = []
+    images_meta = []
+
+    ii = 0
+    for cat, cat_images in image_set.items():
+        for img_name, img in cat_images.items():
+            meta = dict(
+                image_category=cat.decode("utf-8"),
+                image_name=img_name.decode("utf-8"),
+                orientation=np.NaN,
+                phase=np.NaN,
+                spatial_frequency=np.NaN,
+                image_index=ii,
+            )
+
+            images.append(img)
+            images_meta.append(meta)
+
+            ii += 1
+
+    images_dict = dict(
+        metadata=metadata,
+        images=images,
+        image_attributes=images_meta,
+    )
+
+    return images_dict
+
+
+def get_gratings_metadata(stimuli: Dict, start_idx: int = 0) -> pd.DataFrame:
+    """
+    This function returns the metadata for each unique grating that was
+    presented during the experiment. If no gratings were displayed during
+    this experiment it returns an empty dataframe with the expected columns.
+    Parameters
+    ----------
+    stimuli:
+        The stimuli field (pkl['items']['behavior']['stimuli']) loaded
+        from the experiment pkl file.
+    start_idx:
+        The index to start index column
+
+    Returns
+    -------
+    pd.DataFrame:
+        DataFrame containing the unique stimuli presented during an
+        experiment. The columns contained in this DataFrame are
+        'image_category', 'image_name', 'image_set', 'phase',
+        'spatial_frequency', 'orientation', and 'image_index'.
+        This returns empty if no gratings were presented.
+
+    """
+    if "grating" in stimuli:
+        phase = stimuli["grating"]["phase"]
```

Comment:
This might be dangerous in the case that the dictionary is malformed. You can use .get() to avoid issues

```
grating = stimuli.get("grating", {})
phase = grating.get("phase", default_value)
```

Reply:
Noted! I changed the original check to be a get call. Should we be doing this for checks within the if statement as well? I expect that we'd likely want malformed dictionaries to throw errors and fail here, because they wouldn't be following camstim expected standards, but I am not sure what best practice is here.